pkg/validator/http_handler.go

```diff
@@ -0,0 +1,166 @@
+package validator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/filter"
+)
+
+var (
+	scheme       = runtime.NewScheme()
+	codecs       = serializer.NewCodecFactory(scheme)
+	Deserializer = codecs.UniversalDeserializer()
+)
+
+type HttpHandlerFactoryConfig struct {
+	CtrlClient client.Client
+}
+
+type HttpHandlerFactory struct {
+	ctrlClient client.Client
+}
+
+func NewHttpHandlerFactory(config HttpHandlerFactoryConfig) (*HttpHandlerFactory, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+
+	h := &HttpHandlerFactory{
+		ctrlClient: config.CtrlClient,
+	}
+
+	return h, nil
+}
+
+func (h *HttpHandlerFactory) NewCreateHttpHandler(webhookCreateHandler WebhookCreateHandler) http.HandlerFunc {
+	validateFunc := func(ctx context.Context, review v1beta1.AdmissionReview) error {
+		object, err := webhookCreateHandler.Decode(review.Request.Object)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		ok, err := filter.IsCRProcessed(ctx, h.ctrlClient, object.GetObjectMeta())
```

Comment:
Checking if `azure-admission-controller` should validate CR. If this returns `false`, validation is not called at all.

Reply:
Same is done for update below.